lesson18-19/src/pages/login.page.ts

```diff
@@ -0,0 +1,39 @@
+import { Locator, Page } from '@playwright/test';
+import { BasePage } from './base.page';
+
+export class LoginPage extends BasePage {
+    private readonly emailField: Locator;
+    private readonly passwordField: Locator;
+    private readonly loginButton: Locator;
+
+    public constructor(page: Page, baseUrl?: string) {
+        super(page, baseUrl);
+        this.emailField = page.locator('//form[@name="login"]//input[@type="email"]');
+        this.passwordField = page.locator('//form[@name="login"]//input[@type="password"]');
+        this.loginButton = page.locator('//form[@name="login"]//button[@type="submit"]');
+    }
+
+    public async isEmailFieldDisplayed(): Promise<boolean> {
+        return this.emailField.isVisible();
+    }
```

Comment:
використовуйте по максимуму експектейшени плейрайта
https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-visible
і відповідно неймінг методів зі словом експект

```suggestion
    public async expectEmailFieldDisplayed(): Promise<void> {
        await expect(this.emailField).toBeVisible();
    }
```

Reply:
updated methods